DiagnosticsExtension/Models/ConnectionStringValidator/ManagedIdentityCredentialTokenValidator.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ManagedIdentityCredentialTokenValidator.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Web;
+using Azure.Core;
+using Azure.Identity;
+using DiagnosticsExtension.Models.ConnectionStringValidator.Exceptions;
+
+namespace DiagnosticsExtension.Models.ConnectionStringValidator
+{
+    public static class ManagedIdentityCredentialTokenValidator
+    {
+        public static ManagedIdentityCredential GetValidatedCredential(string clientId, string appSettingName)
```

Comment:
I wonder if `appSettingName` parameter is needed here. Can we catch the exception at the callsite of this function and add `appSettingName` into error message there? So that this class can be a more general purpose one and no need to stick to `appSettingName`

Reply:
This function specifically deals with clientid appsetting set in Azure functions only. 
ClientIdInvalidTokenGeneratedSummary - This variable has documentation link related to Azure functions only.
Can we create a new method, which can be more generic?

Reply:
I don't think `clientId` is something specific to Function App, so I believe it can be used for webapp as well without any modification. But I just noticed that the scope was set to `https://storage.azure.com/.default`, so is this function exclusively for storage account? If so, can you update the class name to reflect this?